### ISSUE TITLE
Provide users with consistent success/failure messaging

### ISSUE THREAD

**Issue by briri** [DMPRoadmap/roadmap#175]
Each controller method handles success/failure notifications in different ways. Some do not provide any messaging!

We are now using ActiveRecord validations which place errors into the model's error array. Rails is designed to bubble validation errors up to the UI via this array. We should build a helper method that accepts an ActiveRecord model and will provide a standard success/failure message along with any specific model errors (e.g. 'invalid email address', 'title cannot be blank', etc.)

We should also follow a pattern like this: https://mattbrictson.com/dynamic-rails-error-pages for displaying friendly HTTP 401, 403, 404 and 500 pages.

Consult with UX for styling and placement of these messages.

**Comment by briri** [comment]
related to #173

**Comment by briri** [comment]
@JEK-III and @rhu001 please review the system's current success/failure messages that appear when you try to save changes on different pages. We should provide users with something consistent.

**Comment by briri** [comment]
The error messaging has been consolidated into the following message formats:

If the field was empty or blank and it should not have been the message will be: 
```
Could not [create/update] your [object]
[field] - must not be [blank/empty]

Example:
  Could not update your Plan
  Title - must not be blank.
```

For other types of errors we use a similar format but show the value the user entered instead of the field name:
```
Could not [create/update] your [object]
'[value]' - [message]

Example:
  Could not update your Organization
  'erg334b' - is not a valid email address
```

**Comment by briri** [comment]
Consolidate the success messaging like we did for the error messaging.

Go back through and make sure that errors are being assigned to the `flash[:alert]` and success messages are assigned to `flash[:notice]`. We now display different icons and potentially colors for each one.

**Comment by briri** [comment]
related to #457

**Comment by briri** [comment]
@rhu001 and @JEK-III can you provide us with recommendations for the location and styling of error messages and notifications?

**Comment by rhu001** [comment]
Hi @briri is there a place where you have a listing of some or all of the various messages and notifications currently available so we get a sense of the different lengths of messages and types of messages?

**Comment by briri** [comment]
Hey @rhu001 this isn't an exhaustive list, but it should cover the majority of messages:
```
Information was successfully updated.
Could not create your %{o}.      <--- where 'o' is the object type (e.g. Plan, template, etc.)
Could not update your %{o}.
Could not delete your %{o}.

Organisation was successfully updated.
There seems to be a problem with your logo. Please upload it again.
No institutions are currently registered.
Your institution does not seem to be properly configured.
Please choose an institution

Plan was successfully created. This plan is based on the default template.
Plan was successfully created. This plan is based on the #{plan_params[:funder_name] template with customisations by the plan_params[:org_name]
Plan was successfully created. This plan is based on the')} #{@plan.template.org.name} template.
This is a <strong>test plan</strong>
Guidance choices saved.
Plan was successfully duplicated.

Cannot share plan with %{email} since that email matches with the owner of the plan.
Plan is already shared with %{email}.
Plan shared with %{email}.
Invitation to %{email} issued successfully.
Please enter an email address
Sharing details successfully updated.
Access removed

Export settings updated successfully.
An error has occurred while saving/resetting your export settings.

Your template has been published and is now available to users.
That template is not currently published.
Your template is no longer published. Users will not be able to create new DMPs for this template until you re-publish it
You are viewing a historical version of this template. You will not be able to make changes.
You cannot delete historical versions of this template.

Successfully unlinked your account from %{is}.
Unable to unlink your account from %{is}.
Successfully signed in
Your account has been successfully linked to %{scheme}.
Unable to link your account to %{scheme}.'
The current #{scheme.description} iD has been already linked to a user with email #{identifier.user.email}
```

**Comment by briri** [comment]
I standardized the the success messages to follow this format:
`"Successfully [action] your [object]"` where action and object are specific to the circumstance (e.g. "Successfully saved your example answer.", "Successfully created your template.", "Successfully copied your plan.", "Successfully deleted your question.", etc.)

**Comment by stephaniesimms** [comment]
Additional messaging considerations:
- When I tick the "Test" box on the table in My Dashboard, I get a green success message that says "The" (screenshot 1)
- When I create a new test plan using the default template I get two stacked messages (screenshot 2)

"The"
![the](https://user-images.githubusercontent.com/12703638/28140458-43f08dfe-670d-11e7-91ba-41d07254dcde.png)

Two messages
![two-messages](https://user-images.githubusercontent.com/12703638/28140469-4cf99396-670d-11e7-96f6-df5dfda76056.png)

**Comment by rhu001** [comment]
@briri I think it will be good to have all of the messages be located in the same place but have different background colors. Red for errors and perhaps green for normal notifications. I'll mock something up if you need it.

**Comment by briri** [comment]
Ok @rhu001 here are some examples of the errors that appear at the top of the page (produced by the application after a form submission and the page reloads)
![screen shot 2017-07-13 at 11 02 40 am](https://user-images.githubusercontent.com/1204467/28180866-59b0e2e8-67bb-11e7-9089-542026986b2d.png)
![screen shot 2017-07-13 at 11 04 14 am](https://user-images.githubusercontent.com/1204467/28180865-59b0bd9a-67bb-11e7-9a82-7dfd25ce957e.png)

And here is an example of an inline form validation (trigger by JS after the input field loses focus). And the submit error to let users know why its disabled (trigger by JS when the disabled button receives focus):
![screen shot 2017-07-13 at 10 53 35 am](https://user-images.githubusercontent.com/1204467/28180897-89908f40-67bb-11e7-8638-f353eb364895.png)

**Comment by briri** [comment]
And here's what the inline JS form validation errors look like when we don't have room to display them to the right of the box
![screen shot 2017-07-13 at 11 37 02 am](https://user-images.githubusercontent.com/1204467/28182096-bce32a48-67bf-11e7-9a92-980a6241ff27.png)

**Comment by rhu001** [comment]
@briri Those look good to me. Thanks!

**Comment by stephaniesimms** [comment]
Functional consistency achieved. Styling considerations will be addressed in related issue #424. Closing this issue.
